crates/cheatable-starknet/src/lib.rs

```diff
@@ -20,8 +23,17 @@ impl CheatedState {
             rolled_contracts: HashMap::new(),
             pranked_contracts: HashMap::new(),
             warped_contracts: HashMap::new(),
+            deploy_counter: 0,
         }
     }
+
+    pub fn increment_deploy_counter(&mut self) -> () {
+        self.deploy_counter += 1;
+    }
+
+    pub fn gen_salt(&self) -> ContractAddressSalt {
```

Comment:
I think this could be `generate_salt` (for sake of being explicit), and it could also automatically increment so we don't forget to do it manually.

Reply:
the thing is, that we need to do it separately. Because when we want to precalculate contract_address we won't update `deploy_salt_base`. Maybe I will rename `generate_salt` to `get_salt` 🤔 what do you think? @Arcticae